ReactiveXComponent/WebSocket/WebSocketXCApiManager.cs

```diff
@@ -0,0 +1,113 @@
+
+namespace ReactiveXComponent.WebSocket
+{
+    public class WebSocketXCApiManager
+    {
+        private readonly IWebSocketClient _webSocketClient;
+
+        private event EventHandler<List<string>> XCApiListReceived;
+        private event EventHandler<string> XCApiReceived;
+
+        private readonly IObservable<List<string>> _xcApiListStream;
+        private readonly IObservable<string> _xcApiStream;
+
+        public WebSocketXCApiManager(IWebSocketClient webSocketClient)
+        {
+            _webSocketClient = webSocketClient;
+
+            _xcApiListStream = Observable.FromEvent<EventHandler<List<string>>, List<string>>(
+                handler => (sender, e) => handler(e),
+                h => XCApiListReceived += h,
+                h => XCApiListReceived -= h);
+
+            _xcApiStream = Observable.FromEvent<EventHandler<string>, string>(
+                handler => (sender, e) => handler(e),
+                h => XCApiReceived += h,
+                h => XCApiReceived -= h);
+        }
+
+        public List<string> GetXCApiList(TimeSpan? timeout = null)
+        {
+            var delay = timeout ?? TimeSpan.FromSeconds(10);
+            List<string> result = null;
+            var lockEvent = new AutoResetEvent(false);
+            var observer = Observer.Create<List<string>>(message =>
+            {
+                result = new List<string>(message);
+                lockEvent.Set();
+            });
+
+            using (_xcApiListStream.Subscribe(observer))
+            {
+                _webSocketClient.MessageReceived += ProcessResponse;
+                SendRequest(WebSocketCommand.GetXCApiList);
+                lockEvent.WaitOne(delay);
+                _webSocketClient.MessageReceived -= ProcessResponse;
+            }
+
+            return result;
+        }
+
+
+        public string GetXCApi(string apiFullName, TimeSpan? timeout = null)
+        {
+            var delay = timeout ?? TimeSpan.FromSeconds(10);
+            var result = string.Empty;
+            var lockEvent = new AutoResetEvent(false);
+            var observer = Observer.Create<string>(message =>
+            {
+                result = message;
+                lockEvent.Set();
+            });
+
+            using (_xcApiStream.Subscribe(observer))
+            {
+                _webSocketClient.MessageReceived += ProcessResponse;
+                SendRequest(WebSocketCommand.GetXCApi, apiFullName);
+                lockEvent.WaitOne(delay);
+                _webSocketClient.MessageReceived -= ProcessResponse;
+            }
+
+            return result;
+        }
+
+        private void SendRequest(string requestCommand, string apiFullName = null)
+        {
+            var webSocketRequest = WebSocketMessageHelper.SerializeXCApiRequest(requestCommand, apiFullName);
+            _webSocketClient.Send(webSocketRequest);
+        }
+
+        private void ProcessResponse(object sender, WebSocketMessageEventArgs args)
```

Comment:
Aren't we supposed to create and send a _RequestId_ with our request? Then aren't we supposed to match our _RequestId_ with the one received in the response?